### ISSUE TITLE
How to annotate a function that can take an array consisting of strings and or functions?

### ISSUE THREAD

**Issue by faceleg** [facebook/flow#199]
I am building an interface for AngularJS. My initial work on the interface is below:

```JavaScript
declare class AngularJSModule {
  controller(name: string, dependencies: Array<Function | string>): AngularJSModule;
}

declare class AngularJS {
  module(name: string, dependencies: Array<string>): AngularJSModule ;
}

declare var angular: AngularJS;
```

And some sample code:

```JavaScript
/* @flow */
angular.module('testModule', ['dep1', 'dep2', 'dep3'])
.controller('CapacityProfileViewController', ['$scope', '$window', '$routeParams', '$location', 'modal',
 function($scope, $window, $routeParams, $location, modal) {
}]);
```

The error produced: 

```

example.js:4:2,5:1: function
This type is incompatible with
  flow-interfaces-angular/interfaces/angular.js:6:59,64: string

flow-interfaces-angular/interfaces/angular.js:6:48,55: function type
Too few arguments (expected default/rest parameters in function)
  example.js:4:2,5:1: function

Found 2 errors
```

**Comment by ajwhite** [comment]
:+1: this is a great point

**Comment by faceleg** [comment]
Has this been fixed as part of #256?

**Comment by samwgoldman** [comment]
The original code example no longer produces type errors for me. Please reopen if this is still an issue.
